2/contrib/jenkins/kube-slave-common.sh

```diff
@@ -132,6 +134,76 @@ function generate_kubernetes_config() {
+    "
+
+    if [ -n "$DOTNET10_SLAVE" ]; then
+    echo "
+        <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+            <inheritFrom></inheritFrom>
+            <name>dotnet10</name>
+            <instanceCap>2147483647</instanceCap>
+            <idleMinutes>0</idleMinutes>
+            <label>dotnet10</label>
+            <serviceAccount>${oc_serviceaccount_name}</serviceAccount>
+            <nodeSelector></nodeSelector>
+            <volumes/>
+            <containers>
+              <org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
+                <name>jnlp</name>
+                <image>${DOTNET10_SLAVE}</image>
+                <privileged>false</privileged>
+                <alwaysPullImage>false</alwaysPullImage>
+                <workingDir>/tmp</workingDir>
+                <command></command>
+                <args>\${computer.jnlpmac} \${computer.name}</args>
+                <ttyEnabled>false</ttyEnabled>
+                <resourceRequestCpu></resourceRequestCpu>
+                <resourceRequestMemory></resourceRequestMemory>
+                <resourceLimitCpu></resourceLimitCpu>
+                <resourceLimitMemory></resourceLimitMemory>
+                <envVars/>
+              </org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
+            </containers>
+            <envVars/>
+            <annotations/>
+            <imagePullSecrets/>
+            <nodeProperties/>
+          </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+          <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+            <inheritFrom></inheritFrom>
+            <name>dotnet11</name>
+            <instanceCap>2147483647</instanceCap>
+            <idleMinutes>0</idleMinutes>
+            <label>dotnet11</label>
+            <serviceAccount>${oc_serviceaccount_name}</serviceAccount>
+            <nodeSelector></nodeSelector>
+            <volumes/>
+            <containers>
+              <org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
+                <name>jnlp</name>
+                <image>${DOTNET11_SLAVE}</image>
+                <privileged>false</privileged>
+                <alwaysPullImage>false</alwaysPullImage>
+                <workingDir>/tmp</workingDir>
+                <command></command>
+                <args>\${computer.jnlpmac} \${computer.name}</args>
+                <ttyEnabled>false</ttyEnabled>
+                <resourceRequestCpu></resourceRequestCpu>
+                <resourceRequestMemory></resourceRequestMemory>
+                <resourceLimitCpu></resourceLimitCpu>
+                <resourceLimitMemory></resourceLimitMemory>
+                <envVars/>
+              </org.csanchez.jenkins.plugins.kubernetes.ContainerTemplate>
+            </containers>
+            <envVars/>
+            <annotations/>
+            <imagePullSecrets/>
+            <nodeProperties/>
+          </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+    "
```

Comment:
adding this to the default config for jenkins is problematic since it's going to get added to the config for the jenkins-centos image also, but there's no .net centos image available(?) which means either users who are using the centos image are going to pick up rhel slave images (despite not having valid subscriptions) or this is going to try to pull something that doesn't exist (depending how you define DOTNET11_SLAVE and DOTNET10_SLAVE)

That said, we need to discuss whether these should be defined by default anyway, or we should just let users pick them up, we've made picking up slave images dynamically much easier in 3.6:
https://docs.openshift.org/latest/using_images/other_images/jenkins.html#using-the-jenkins-kubernetes-plug-in-to-run-jobs